kafkabp/consumer_interface.go

```diff
@@ -25,20 +29,25 @@ type ConsumerConfig struct {
 	// Required. Topic is used to specify the topic to consume.
 	Topic string `yaml:"topic"`
 
-	// Required. ClientID is a user-provided string sent with every request to
-	// the brokers for logging, debugging, and auditing purposes. The default
-	// Consumer implementation in this library expects every Consumer to have a
-	// unique ClientID.
+	// Required. ClientID is used by Kafka broker to track clients' consuming
+	// progresses on the topics.
 	//
+	// For NewConsumer, every instance is expected to have a unique ClientID.
 	// The Kubernetes pod ID is usually a good candidate for this unique ID.
+	//
+	// For NewGroupConsumer, instances from the same group must share the same
+	// ClientID.
 	ClientID string `yaml:"clientID"`
```

Comment:
Should this become `GroupID` whereas I believe `ClientID` is used to distinguish single requests? The client ID here is at least passed into the sarama implementation which declares it as group ID too.

Reply:
Good point. Added `GroupID` to `ConsumerConfig` instead, this way we don't even need to export `NewGroupConsumer`, and `NewConsumer` can be smart enough to choose the implementation automatically.